src/handlers.cpp

```diff
@@ -1578,29 +1640,120 @@ void ImpuRegDataTask::on_sar_response(Diameter::Message& rsp)
       {
         TRC_DEBUG("Getting User-Data from SAA for cache");
         saa.user_data(_xml);
-        put_in_cache();
+        caching = true;
```

Comment:
Yes, I believe so - what was your concern?

*   The AVP shouldn't be present on deregistration or authentication failure responses.
*   If the AVP is present, we return it on the HTTP response.
*   If the AVP is not present, we omit it from the response (as at present).